app/routes/surveys/AddSurveyRoute.js

```diff
@@ -1,31 +1,85 @@
 import { connect } from 'react-redux';
 import { compose } from 'redux';
-import { addSurvey, deleteSurvey } from '../../actions/SurveyActions';
+import {
+  addSurvey,
+  deleteSurvey,
+  fetchTemplate
+} from '../../actions/SurveyActions';
 import SurveyEditor from './components/SurveyEditor/SurveyEditor';
 import { LoginPage } from 'app/components/LoginForm';
 import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
 import { push } from 'react-router-redux';
-import moment from 'moment-timezone';
 import { initialQuestion } from './components/SurveyEditor/SurveyEditor';
+import prepare from 'app/utils/prepare';
+import { selectSurveyTemplate } from 'app/reducers/surveys';
+import { fetchEvent } from 'app/actions/EventActions';
+import { selectEventById } from 'app/reducers/events';
+import { defaultActiveFrom } from './utils';
 
-const time = (hours, minutes) =>
-  moment()
-    .startOf('day')
-    .add({ hours, minutes })
-    .toISOString();
+const loadData = (props, dispatch) => {
+  const { template, event } = props.location.query;
+  if (event) {
+    return dispatch(fetchEvent(event)).then(
+      result =>
+        result.success &&
+        dispatch(fetchTemplate(result.payload.entities.events[event].eventType))
+    );
+  }
+  if (template) {
+    return dispatch(fetchTemplate(template));
+  }
+};
 
 const mapStateToProps = (state, props) => {
+  const notFetching = !state.surveys.fetching && !state.events.fetching;
+  const { templateType, event } = props.location.query;
+
+  const fullEvent = selectEventById(state, { eventId: event });
+
+  const template = selectSurveyTemplate(state, {
+    ...props,
+    templateType: templateType || fullEvent.eventType
+  });
+
+  const initialEvent = event
```

Comment:
why can an event be both an object and a string?

Reply:
I think that's how the `SelectInput` autocomplete thing works? Sending an empty string makes it empty, while sending an object on the form
```
{
        value: fullEvent.id,
        label: fullEvent.title
}
```
makes it select that event by default. I suppose that instead of an empty string I could send
```
{
        value: '',
        label: ''
}
```
I just didn't see the need or advantage